app/cdap/components/GridTable/index.tsx

```diff
@@ -232,10 +182,16 @@ export default function GridTable() {
   return (
     <Box data-testid="grid-table">
       <BreadCrumb datasetName={wid} />
+      {Array.isArray(gridData?.headers) && gridData?.headers.length === 0 && (
+        <NoRecordScreen
```

Comment:
null check

Reply:
Not required

Reply:
it is already getting checked in this (Array.isArray(gridData?.headers) ) condition, So we don't have to explicitly check again for headers null.